### ISSUE TITLE
Random Crashes

### ISSUE THREAD

**Issue by CecilFF4** [libretro/beetle-psx-libretro#191]
So I get random crashes every now and then, no matter what game I'm playing. Here's my setup:
Win10 x64
Retroarch 1.5.0
Beetle PSX HW 0.9.44.1 76862ab
Windowed full-screen.
16:9 aspect ratio.
Vsync on.
Renderer: Vulkan
Internal resolution: any value (I tried various and got the same results)
Skip BIOS: On
All other values set to default.

For instance, I'll start up FF7. It goes along nicely for a time; fight a few battles, traverse a few screens. Then I start to get audio stuttering and a drop in the framerate. Sometimes it corrects itself for a short time. Other times it'll freeze and crash. This sequence takes about 5-10 minutes. I can't force it to crash; it just does it when it feels like it. This is not the only game this happens on.

**Comment by omega64** [comment]
For me HW version crash too, specially if i use vulkan.
For what i can say, i have just finish yesterday Metal Gear Solid Disc 1 & 2 with no issues.
- Win10 x64
- RetroArch 1.5.0
- Beetle PSX
- Windowed Full-Screen
- 4:3 Aspect Ratio
- Vsync On
- Render: OpenGL
- Internal Resolution: Native (if i change, it crash)
- Skip BIOS: No
- Widescreen Hack: No
- No hacks or special settings, all defaults.
(I know this isn't the fancy graphics options, but i like native rendering)

Try with this configuration and see if you can play normally.

**Comment by CecilFF4** [comment]
SW version of Beetle is too slow for me. Sticking with HW, I discovered:

Renderer: irrelevant
Aspect Ratio: irrelevant
Vsync: irrelevant
Skip BIOS: irrelevant

No changes that I implemented based on your instructions alleviated the issue, unfortunately. If you wouldn't mind, though, could you try playing FF7 (if you have it) for 10 min or so and see if the crash happens for you? FF8 is another game that crashes for me if you'd like another option. If you can, thank you.

**Comment by CecilFF4** [comment]
So I've verified this issue with games other than FF7 and FF8. It's certainly game-independent. I was also finally able to get it to output a log file (attached). Lastly, here's a little more information about my setup that may be useful:

Launch Steam BPM with Steam controller connected. Load up controller configuration (outputs as xinput).
Play Launchbox (external shortcut).
Launch FF7 using Retroarch emulator with command-line options: -L "cores\mednafen_psx_hw_libretro.dll" -c "config\psx_generic.cfg". (attached config and core config)
Using m3u playlist pointing at cue/bins.
Saves and loads states just fine.

[log.txt](https://github.com/libretro/beetle-psx-libretro/files/1034876/log.txt)
[psx_generic.cfg.txt](https://github.com/libretro/beetle-psx-libretro/files/1034942/psx_generic.cfg.txt)
[Final Fantasy 7.opt.txt](https://github.com/libretro/beetle-psx-libretro/files/1034943/Final.Fantasy.7.opt.txt)

**Comment by CecilFF4** [comment]
I have crashes on other cores, too. Does no one else have this problem? Can anyone help?

**Comment by twinaphex** [comment]
Each core is its own separate program. They are not perfect, and they often have memory leaks and other unresolved issues which aren't even our fault if they were not written by us.

I have no idea about your crashes. I don't get them here.

**Comment by CecilFF4** [comment]
Sure, but if multiple cores are having the same issues, wouldn't that imply there's a problem with RA or a non-core setting?

**Comment by twinaphex** [comment]
No it wouldn't. I think as a non-developer you shouldn't imply anything really.

**Comment by CecilFF4** [comment]
No, I'm not a developer. I just used simple logic. Which makes more sense: every core I have has something wrong with it or a setting somewhere in RA needs to be tweaked? Now, I'm not trying to be aggressive here. I just have a crashing issue that happens consistently and can be easily reproduced on my computer using the settings I've posted. Yes, it happens more often on beetle. The one time desmume failed could've just been a coincidence; I felt it was worth mentioning. But if we disregard that, the crash still happens all the time on at least this one core (beetle). It may or may not be related to the core itself. All I'm asking for is help from someone that could review the settings and log file I've posted. Please.

**Comment by twinaphex** [comment]
It's worth mentioning but it's also worth remembering that we are short on resources and that not every emulator we have a core for is even our own creation. You are asking for a level of perfection and stability that you wouldn't even get with the 'standalone' offering.

Also, badgering the few developers who can contribute to these cores is not going to help your case either. So please try to stay respectful and quit being impatient and jumpy.

**Comment by CecilFF4** [comment]
I'm not asking for perfection. Just a solution to the crashes, even if it's some ugly workaround. This post is up more to see if this is a widespread issue, because if it was, it would get the attention it deserves. But since no one else is posting about their RA crashing, then I totally understand resources can't be diverted to this. Sorry you feel that asking for help and waiting for days for any kind of response (even if the response is "we can't help right now") is "badgering" and "impatient." I never meant any disrespect. Again, sorry you feel that way.

**Comment by twinaphex** [comment]
I can't fix crashes I cannot reproduce.

**Comment by saftle** [comment]
@CecilFF4 grab the newest nightly build and see if the crash is still reproducible, also be sure to post the GIT version of the nightly you're using. It might help isolate a specific nightly (if the crash is perhaps reproducible only with a certain build). Most developers are on a nightly and not on the last stable release, since the nightlies are pretty stable most of the time.

**Comment by CecilFF4** [comment]
Sorry it's been a while since I could test, but I got build 09d2a02 and the crash still happens. "Retroarch has stopped working" is all that shows up when I've played for a few minutes. However, I was able to figure out that if EITHER internal resolution is upped OR aspect ratio is changed from default, the crash happens. If BOTH of those are on default settings, the crashes don't seem to happen (as far as I can tell). The whole reason I'm playing RA, though, is because I want to play FF7 in a higher resolution. :(

**Comment by saftle** [comment]
@CecilFF4 interesting. Try this quote that I copied from libretro.com:

> To get error logs, run retroarch_debug.exe --menu --verbose from a command line and it will print things to the console window. Sometimes you need to run it through gdb to get info on more mysterious crashes.

A bit off-topic, but as an FF7 enthusiast I would recommend FF7 on PC (Steam) since it has TONS of amazing mods to choose from and can be easily downloaded using the 7th Heaven mod manager: http://forums.qhimm.com/index.php?topic=15520.0

**Comment by CecilFF4** [comment]
Here's another log from start til crash. Also, I appreciate the advice to get FF7 on Steam, but I want my PS1 ultimately working for all my PS1 games; I just got the FF7-itch at the moment, but FF8, FF9, Colony Wars, G-Darius, Wipeout XL, and more have nostalgic value and I'd like to play them all at some point.

[log.txt](https://github.com/libretro/beetle-psx-libretro/files/1093527/log.txt)

**Comment by CecilFF4** [comment]
Just ignore everything I said previously.

I started with fresh configs (standard and core) and only changed the renderer to Vulkan (so no enhanced resolution or altered aspect ratio). It crashed after about 4-5 minutes, consistently. Changed the renderer to GL. Worked. Now, this renderer setting is the one in the driver menu, not the core menu, so this problem is probably posted in the wrong place (but anyone reading this still deserves to know how this went). I was able to get Vulkan to work if I didn't do windowed fullscreen, btw (though I get flickering during throttle and a zoomed RA menu screen). I also tried the x86 version of RA, but Steam didn't recognize the game as a "game" when using Vulkan (if I can't use my controller, what's the point, right?). Lastly, RA with Vulkan in windowed fullscreen doesn't seem to crash when using other cores, but it could be that the length of time to crash is longer.

FYI, I think the reason it crashed after changing the renderer on May 29th was because I changed the core renderer instead of the driver menu one.

TL;DR - My previous assumptions were incorrect: it's the driver menu renderer that was causing issues. If using non-windowed fullscreen, I can use Vulkan. If using windowed fullscreen, I can only use GL.

**Comment by rz5** [comment]
@CecilFF4, several bugs have been fixed that would cause beetle-psx and RA to crash when changing from windowed to fullscreen and when using core options like dithering/upscaling/texture filtering.

Try it all, and if it's all fixed, remember to close this issue.
